src/manager/addons.js

```diff
@@ -0,0 +1,68 @@
+import { Svue } from "svue";
+import {
+  // getAddon,
+  getAddons,
+  postAddonDispatch,
+} from "@/api/addon";
+
+export const addons = new Svue({
+  data() {
+    return {
+      addons: [],
+    };
+  },
+  computed: {
+    addonsById(addons) {
+      const results = {};
+      for (let i = 0; i < addons.length; i++) {
+        const addon = addons[i];
+        results[addon.id] = addon;
+      }
+      return results;
+    }
+  },
+});
+
+
+export async function initAddons(me) {
+  const newAddons = await getAddons();
+
+  addons.addons = newAddons;
+}
+
+export async function dispatchAddon(addonId, info,
```

Comment:
I don't think `layout` or `viewerUpdate` are needed here.  We can decide if we want to be able to call Add-Ons from the viewer at some point and if we should leave that code in place for that reason.